lib/acts_as_list/active_record/acts/list.rb

```diff
@@ -237,7 +247,7 @@ def add_to_list_top
           if not_in_list? || internal_scope_changed? && !position_changed || default_position?
             increment_positions_on_all_items
             self[position_column] = acts_as_list_top
-          else
+          elsif position_in_list?(self[position_column], id)
```

Comment:
You don't need to pass in the first argument because it's available within the method call already since it's an instance method of `self`. To this end, you could rename the method `position_collides_with?(id)` or something of that nature. Inside the method just call `send(:position_column)` to get the position value of the current instance. Of you can use the array syntax, but just check to see what we're doing in the rest of the code. I can't remember what convention we were using :)